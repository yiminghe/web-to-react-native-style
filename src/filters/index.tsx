import border from './border';
import borderRadius from './borderRadius';
import borderColor from  './borderColor';
import display from './display';
import paddingMargin from './paddingMargin';
import position from './position';
import string from './string';
export default [
  string,
  border,
  borderRadius,
  borderColor,
  display,
  paddingMargin,
  position,
] as any;
