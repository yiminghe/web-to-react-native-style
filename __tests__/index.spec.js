import { transform } from '../src/';

describe('web-to-react-native-style', () => {
  it('support padding', () => {
    expect(transform('padding', '10px 4px')).toMatchSnapshot();
  });

  it('support margin', () => {
    expect(transform('margin', '10px 4px')).toMatchSnapshot();
  });

  it('support font-weight', () => {
    expect(transform('font-weight', '10')).toMatchSnapshot();
  });

  it('support border', () => {
    expect(transform('border', '10px solid red')).toMatchSnapshot();
  });

  it('support border-radius', () => {
    expect(transform('border-radius', '1px 2px')).toMatchSnapshot();
  });

  it('support border-color', () => {
    expect(transform('border-color', 'red #f015ca')).toMatchSnapshot();
  });

  it('support border-width', () => {
    expect(transform('border-width', '2px 5rpx')).toMatchSnapshot();
  })

  it('support background', () => {
    expect(transform('background', '#fff')).toMatchSnapshot();
  });

  it('support box-shadow', () => {
    expect(transform('box-shadow', '10px 5px 5px black;')).toMatchSnapshot();
  });

  it('support text-shadow', () => {
    expect(transform('text-shadow', '10px 5px 5px black;')).toMatchSnapshot();
  });

  it('support font-family', () => {
    expect(transform('font-family', `'lucida grande', tahoma, verdana, arial, sans-serif`)).toMatchSnapshot();
  });

  it('support transform', () => {
    expect(transform('transform', `translate(12px, 50%) scale(2, 0.5) rotate(10deg)`)).toMatchSnapshot();
  });

  it('support flex', () => {
    expect(transform('flex', '2')).toMatchSnapshot();
    expect(transform('flex', '1 2')).toMatchSnapshot();
    expect(transform('flex', '1 0 2')).toMatchSnapshot();
  });
});
