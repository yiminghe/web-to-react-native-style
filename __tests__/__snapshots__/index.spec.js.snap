// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`web-to-react-native-style support background 1`] = `
Object {
  "backgroundColor": "#fff",
}
`;

exports[`web-to-react-native-style support border 1`] = `
Object {
  "borderColor": "red",
  "borderStyle": "solid",
  "borderWidth": 10,
}
`;

exports[`web-to-react-native-style support border-color 1`] = `
Object {
  "borderBottomColor": "red",
  "borderLeftColor": "#f015ca",
  "borderRightColor": "#f015ca",
  "borderTopColor": "red",
}
`;

exports[`web-to-react-native-style support border-radius 1`] = `
Object {
  "borderBottomLeftRadius": 2,
  "borderBottomRightRadius": 1,
  "borderTopLeftRadius": 1,
  "borderTopRightRadius": 2,
}
`;

exports[`web-to-react-native-style support border-width 1`] = `
Object {
  "borderBottomWidth": 2,
  "borderLeftWidth": 5,
  "borderRightWidth": 5,
  "borderTopWidth": 2,
}
`;

exports[`web-to-react-native-style support box-shadow 1`] = `
Object {
  "overflow": "visible",
  "shadowColor": "black;",
  "shadowOffset": Object {
    "height": 5,
    "width": 10,
  },
  "shadowOpacity": 1,
  "shadowRadius": 5,
}
`;

exports[`web-to-react-native-style support flex 1`] = `
Object {
  "flex": 2,
}
`;

exports[`web-to-react-native-style support flex 2`] = `
Object {
  "flexGrow": 1,
  "flexShrink": 2,
}
`;

exports[`web-to-react-native-style support flex 3`] = `
Object {
  "flexBasis": 2,
  "flexGrow": 1,
  "flexShrink": 0,
}
`;

exports[`web-to-react-native-style support font-family 1`] = `
Object {
  "fontFamily": "lucida grande",
}
`;

exports[`web-to-react-native-style support font-weight 1`] = `
Object {
  "fontWeight": "10",
}
`;

exports[`web-to-react-native-style support margin 1`] = `
Object {
  "marginBottom": 10,
  "marginLeft": 4,
  "marginRight": 4,
  "marginTop": 10,
}
`;

exports[`web-to-react-native-style support padding 1`] = `
Object {
  "paddingBottom": 10,
  "paddingLeft": 4,
  "paddingRight": 4,
  "paddingTop": 10,
}
`;

exports[`web-to-react-native-style support text-shadow 1`] = `
Object {
  "textShadowColor": "black;",
  "textShadowOffset": Object {
    "height": 5,
    "width": 10,
  },
  "textShadowRadius": 5,
}
`;

exports[`web-to-react-native-style support transform 1`] = `
Object {
  "transform": Array [
    Object {
      "translateX": 12,
    },
    Object {
      "translateY": "50%",
    },
    Object {
      "scaleX": 2,
    },
    Object {
      "scaleY": 0.5,
    },
    Object {
      "rotateZ": "10deg",
    },
  ],
}
`;
